### PR TITLE
✨ Merge in CAPD with a single squashed commit

### PR DESCRIPTION
**Pull request by chuckha**
This PR merges in CAPD with a single commit for the CAPD history.

Attribution remains in the (soon to be) archived project.

### CODE REVIEW

**Comment by ncdc** [comment]
Do we need to get rid of the aliases `cluster-api-provider-docker-maintainers` and `cluster-api-provider-docker-admins`? Or add them in?

**Comment by chuckha** [comment]
I'm not sure, I'm looking into it right now

**Comment by detiber** [comment]
> Do we need to get rid of the aliases cluster-api-provider-docker-maintainers and cluster-api-provider-docker-admins? Or add them in?

I believe the ALIASES will need to be added to the root OWNERS_ALIASES file and the embedded OWNERS_ALIASES file will need to be removed

**Comment by chuckha** [comment]
@detiber Yep, what you said looks right according to https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md

**Comment by erwinvaneyk** [comment]
Fully agree with merging CAPD into CAPI - but to give a bit of feedback: since CAPD is targeting both testing and development use cases, as well as it being the reference implementation, wouldn't a location outside `test` be more appropriate?

**Comment by ncdc** [comment]
Master is open - I assume we still want to proceed with this? Are there commits to capd that we need to copy in to this PR?

**Comment by ncdc** [comment]
@erwinvaneyk it's a reference implementation to ensure our e2e/integration tests pass, but I wouldn't call it a reference implementation for people to actually use, would you?

**Comment by erwinvaneyk** [comment]
> @erwinvaneyk it's a reference implementation to ensure our e2e/integration tests pass, but I wouldn't call it a reference implementation for people to actually use, would you?

@ncdc ah okay, then I misunderstood what you mean with "reference implementation". It is indeed  unsuitable for any form of production use, but to me its usefulness stretches beyond test infra (development, code referencing). Which is why I thought that it could do with a bit more prominent location, outside of `test`. In any case, it's a minor detail... 😉

**Comment by vincepri** [comment]
LGTM pending boilerplate fix and OWNERS file

File: test/infrastructure/docker/PROJECT [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: "2"
+domain: cluster.x-k8s.io
+repo: sigs.k8s.io/cluster-api/test/infrastructure/docker
```
Comment (detiber) [inline test/infrastructure/docker/PROJECT]: Do we know if the kubebuilder tooling currently work with this configuration? If not, we should likely file a tracking ticket

File: test/infrastructure/docker/SECURITY_CONTACTS [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Defined below are the security contacts for this repo.
```
Comment (detiber) [inline test/infrastructure/docker/SECURITY_CONTACTS]: We can likely remove this in favor of the SECURITY_CONTACTS at the root of the repo.

Comment (chuckha) [reply test/infrastructure/docker/SECURITY_CONTACTS]: ack

**Comment by chuckha** [comment]
@vincepri updated the go module in a new commit
@detiber removed sec contacts as part of the files removal commit

**Comment by chuckha** [comment]
/hold cancel

#1096 has merged

**Comment by ncdc** [comment]
Waiting on #1515

File: test/infrastructure/docker/PROJECT (hunk 1 above)
Comment (chuckha) [reply test/infrastructure/docker/PROJECT]: https://github.com/kubernetes-sigs/cluster-api/issues/1526

**Comment by chuckha** [comment]
@vincepri @ncdc Anything left before this can merge?

**Pull request merged**